app/src/main/java/net/gsantner/markor/activity/FilesystemListFragment.java

```diff
@@ -262,6 +256,93 @@ public void clearSearchFilter() {
         reloadAdapter();
     }
 
+    public void sortByName(){
+        int size = _filesCurrentlyShown.size();
```

Comment:
You don't need to do this all by hand! Nice work, but you don't need to that all manually, wow :D. There is already a lot of stuff in java standard lib that does this hard work for you. Just use  for example

```
Collections.sort(data, new Comparator(){

compareto(){
 return something.compareto(somethinelse)
}
```

Reply:
Thanks for the info. I will make the necessary changes.